### DISCUSSION TITLE
tulip.text() is wonky

### DISCUSSION THREAD

**Discussion by drepetto** [shorepine/tulipcc#85] [General]
I'm playing with the buttons.py demo. It took me awhile to realize that the tulip.text() boxes are text inputs. A few ideas:

* there should be a cursor or > or prompt of some sort to make it clearer that's it's for text entry
* maybe the API name should be tulip.text_entry()?
* long text just overflows the box. it should scroll or something?

**Comment by bwhitman** [comment]
`text()` definitely could use some work. 

Cursor sounds very hard. We currently highlight the box when it's active. I would hope people just got used to that. Certainly a `>` character is a good idea. 

Don't have any religion on api name. I agree `str` should probably be `text` and `text` could be `text_input` or something, yeah. 

For long text, my general opinion is "well, stop doing that". Scrolling sounds very hard and annoying. I'm not sure how that would work, UI wise. But happy to hear from others on what should happen.

**Reply by bwhitman** [reply]
Or maybe we do it like `tulip.ui.text()` and `tulip.bg.text()` ? Would save us future namespace headaches

**Reply by drepetto** [reply]
Re: naming -- many systems have a text() call that is for drawing labels and whatnot. So I think text_input() or something similar would be better for input boxes. I agree about the namespace split, although I'd still make it tulip.ui.text_input(). I think. maybe it's text() vs label(), I don't know. but I think. non-interactive UI text is needed, and is different from bg.text() or bg.str() or whatever.

**Reply by drepetto** [reply]
Re: long text -- I think it's fine if it just stops accepting text after the char limit, but obvs it shouldn't crash the system if you keep typing...

**Reply by bwhitman** [reply]
yeah. `ui.c` `ui_text_entry_update` should check before trying to add a new character when `len >= UI_TEXT_MAX_LEN`
